src/tracked_unique.rs

```diff
@@ -0,0 +1,152 @@
+//! A small change tracker to wrap around a unique so you can determine if the value changes between updates.
+use tracing::trace_span;
+
+use crate::prelude::*;
+
+use core::any::type_name;
+use std::{fmt, ops::Deref, ops::DerefMut};
+
+pub struct TrackedValue<T: 'static>(InnerTrackedState, T);
+
+pub struct TrackedMut<'a, T: 'static>(UniqueViewMut<'a, TrackedValue<T>>);
+pub struct Tracked<'a, T: 'static>(UniqueView<'a, TrackedValue<T>>);
+
+impl<'a, T: 'static + Send + Sync> Borrow<'a> for Tracked<'a, T> {
+    fn borrow(
+        all_storages: &'a AllStorages,
+        all_borrow: Option<SharedBorrow<'a>>,
+    ) -> Result<Self, error::GetStorage>
+    where
+        Self: Sized,
+    {
+        Ok(Tracked(Borrow::borrow(all_storages, all_borrow)?))
+    }
+}
+
+unsafe impl<'a, T: 'static + Send + Sync> BorrowInfo for Tracked<'a, T> {
+    fn borrow_info(mut info: &mut Vec<info::TypeInfo>) {
+        UniqueView::<'a, T>::borrow_info(&mut info);
+    }
+}
+
+impl<'a, T: 'static + Send + Sync> Borrow<'a> for TrackedMut<'a, T> {
+    fn borrow(
+        all_storages: &'a AllStorages,
+        all_borrow: Option<SharedBorrow<'a>>,
+    ) -> Result<Self, error::GetStorage>
+    where
+        Self: Sized,
+    {
+        Ok(TrackedMut(Borrow::borrow(all_storages, all_borrow)?))
+    }
+}
+
+unsafe impl<'a, T: 'static + Send + Sync> BorrowInfo for TrackedMut<'a, T> {
+    fn borrow_info(mut info: &mut Vec<info::TypeInfo>) {
+        UniqueViewMut::<'a, T>::borrow_info(&mut info);
+    }
+}
+
+#[derive(PartialEq)]
+enum InnerTrackedState {
+    New,
+    Modified,
+    NoChanges,
+}
+
+impl<T> TrackedValue<T> {
+    pub(crate) fn new(value: T) -> Self {
+        TrackedValue(InnerTrackedState::New, value)
+    }
+    fn reset_tracking(&mut self) {
+        self.0 = InnerTrackedState::NoChanges;
+    }
+    fn is_new_or_modified(&self) -> bool {
+        self.0 != InnerTrackedState::NoChanges
+    }
+}
+
+impl<T> Tracked<'_, T> {
+    /// You may only check if Tracked is new or modified for now.
+    pub fn is_new_or_modified(&self) -> bool {
+        self.0.is_new_or_modified()
+    }
+}
+
+impl<T: 'static> Deref for Tracked<'_, T> {
+    type Target = T;
+
+    #[inline(always)]
+    fn deref(&self) -> &T {
+        &self.0 .1
```

Comment:
load-bearing whitespace? Is some parser confused and thinks this was a float? rustc would accept `&self.0.1`.

Reply:
`cargo fmt` / `rustfmt` did this, I think.

Reply:
[This is the explanation](https://github.com/rust-lang/rustfmt/issues/4355#issuecomment-723764276)